etna/models/nn/deepar_new/deepar.py

```diff
@@ -0,0 +1,367 @@
+        decoder_target = batch["decoder_target"].float()  # (batch_size, decoder_length, 1)
+        weights = batch["weight"]
+        target = torch.cat((encoder_target, decoder_target), dim=1)  # (batch_size, encoder_length+decoder_length-1, 1)
+        output, (_, _) = self.rnn(
+            torch.cat((encoder_real, decoder_real), dim=1)
+        )  # (batch_size, encoder_length+decoder_length-1, hidden_size)
+        loc, scale = self.get_distribution_params(output)  # (batch_size, encoder_length+decoder_length-1, 1)
+        target_prediction = self.loss.sample(loc, scale, weights, n_samples=1)
+        loss = self.loss(target, loc, scale, weights)
+        return loss, target, target_prediction
+
+    def make_samples(self, df: pd.DataFrame, encoder_length: int, decoder_length: int) -> Iterator[dict]:
+        """Make samples from segment DataFrame."""
+        segment = df["segment"].values[0]
+        values_target = df["target"].values
+        values_real = (
+            df.select_dtypes(include=[np.number])
+            .assign(target_shifted=df["target"].shift(1))
+            .drop(["target"], axis=1)
+            .pipe(lambda x: x[["target_shifted"] + [i for i in x.columns if i != "target_shifted"]])
+            .values
+        )
+
+        def _make(
+            values_real: np.ndarray,
+            values_target: np.ndarray,
+            segment: str,
+            start_idx: int,
+            encoder_length: int,
+            decoder_length: int,
+        ) -> Optional[dict]:
+
+            sample: Dict[str, Any] = {
+                "encoder_real": list(),
+                "decoder_real": list(),
+                "encoder_target": list(),
+                "decoder_target": list(),
+                "segment": None,
+                "weight": None,
+            }
+            total_length = len(values_target)
+            total_sample_length = encoder_length + decoder_length
+
+            if total_sample_length + start_idx > total_length:
+                return None
+
+            # Get shifted target and concatenate it with real values features
+            sample["decoder_real"] = values_real[start_idx + encoder_length : start_idx + total_sample_length].copy()
+
+            # Get shifted target and concatenate it with real values features
+            sample["encoder_real"] = values_real[start_idx : start_idx + encoder_length].copy()
+            sample["encoder_real"] = sample["encoder_real"][1:]
+
+            target = values_target[start_idx : start_idx + total_sample_length].reshape(-1, 1).copy()
+            sample["encoder_target"] = target[1:encoder_length].copy()
+            sample["decoder_target"] = target[encoder_length:].copy()
+
+            sample["segment"] = segment
+            sample["weight"] = 1 + sample["encoder_target"].mean() if self.scale else 1
+            sample["encoder_real"][:, 0] /= sample["weight"]
```

Comment:
Now you copy all the features, I guess we should create only the copy of shifted_target, so we can do sth like
```
 sample["encoder_real"][:, 0] = values_real[start_idx : start_idx + encoder_length, 0] / sample["weight"]
....
```